br/pkg/streamhelper/basic_lib_for_test.go

```diff
@@ -635,3 +650,97 @@ func (t *testEnv) unregisterTask() {
 		Name: "whole",
 	}
 }
+
+func (t *testEnv) ScanLocksInOneRegion(bo *tikv.Backoffer, key []byte, maxVersion uint64, limit uint32) ([]*txnlock.Lock, *tikv.KeyLocation, error) {
+	for _, r := range t.regions {
+		if len(r.locks) != 0 {
+			return r.locks, &tikv.KeyLocation{
+				Region: tikv.NewRegionVerID(r.id, 0, 0),
+			}, nil
+		}
+	}
+	return nil, nil, nil
+}
+
+func (t *testEnv) ResolveLocksInOneRegion(bo *tikv.Backoffer, locks []*txnlock.Lock, loc *tikv.KeyLocation) (*tikv.KeyLocation, error) {
+	for _, r := range t.regions {
+		if loc != nil && loc.Region.GetID() == r.id {
+			// reset locks
+			r.locks = nil
+			return t.resolveLocks(locks, loc)
+		}
+	}
+	return nil, nil
+}
+
+func (t *testEnv) Identifier() string {
+	return "advance test"
+}
+
+func (t *testEnv) GetStore() tikv.Storage {
+	// only used for GetRegionCache once in resolve lock
+	return &mockTiKVStore{regionCache: tikv.NewRegionCache(&mockPDClient{})}
+}
+
+type mockKVStore struct {
+	kv.Storage
+}
+
+type mockTiKVStore struct {
+	mockKVStore
+	tikv.Storage
+	regionCache *tikv.RegionCache
+}
+
+func (s *mockTiKVStore) GetRegionCache() *tikv.RegionCache {
+	return s.regionCache
+}
+
+func (s *mockTiKVStore) SendReq(bo *tikv.Backoffer, req *tikvrpc.Request, regionID tikv.RegionVerID, timeout time.Duration) (*tikvrpc.Response, error) {
+	scanResp := kvrpcpb.ScanLockResponse{
+		// we don't need mock locks here, because we already have mock locks in testEnv.Scanlocks.
+		// this behaviour is align with gc_worker_test
+		Locks:       nil,
+		RegionError: nil,
+	}
+	return &tikvrpc.Response{Resp: &scanResp}, nil
+}
+
+type mockPDClient struct {
+	pd.Client
+}
+
+func (p *mockPDClient) ScanRegions(ctx context.Context, key, endKey []byte, limit int) ([]*pd.Region, error) {
+	return []*pd.Region{
+		newMockRegion(1, []byte("1"), []byte("3")),
+	}, nil
+}
+
+func (p *mockPDClient) GetStore(_ context.Context, storeID uint64) (*metapb.Store, error) {
+	return &metapb.Store{
+		Id:      storeID,
+		Address: fmt.Sprintf("127.0.0.%d", storeID),
+	}, nil
+}
+
+func (p *mockPDClient) GetRegion(ctx context.Context, key []byte, opts ...pd.GetRegionOption) (*pd.Region, error) {
```

Comment:
Why not delegate those `GetRegion`s to `fakeCluster`? (Just fetch the region info by `fakeCluster.RegionScan` and wrap the returned region with `pd.Region`) It seems we still need to write another `mockPDClient` once we want to write a case which tests two (or more regions) has unlocked txn.